### PR TITLE
fix: Type definition promise wrappers

### LINKED ISSUE #9: Type definitions are incorrect since #8
#8 added a wrapper object to responses returned in the callback and in the promise, but the type definitions were not updated to reflect this change.

Example
```ts
testrail.getRun(1).then(run => {
  console.log(run.id) // Type definitions say `id` exists, but logs `undefined`
  console.log(run.body.id) // Type error, but logs `1`
})
```

### PR DESCRIPTION
**Pull request by NicholasBoll**
Fixes #9

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,92 +1,95 @@
...
+
+    getUser(id: number, callback?: t.Callback<t.ITestrailUser>): t.PromiseResponse<t.ITestrailUser>;
+    getUserByEmail(email: string, callback?: t.Callback<t.ITestrailUser>): t.PromiseResponse<t.ITestrailUser>;
+    getUsers(callback?: t.Callback<t.ITestrailUser[]>): t.PromiseResponse<t.ITestrailUser[]>;
 }
 
 declare namespace TestrailApiClient {
     type CustomFieldType = boolean | string | number | number[] | any[];
 
+    type PromiseResponse<T> = Promise<{ response: Response, body: T }>
```
Comment (NicholasBoll) [inline index.d.ts:91]: The docs calls this a `result`. Perhaps the types should reflect that as well:

```ts
type Result<T> = Promise<{ response: Response, body: T }>
```

**Comment by NicholasBoll** [comment]
@rundef I'm finding a few small issues I'm fixing in my local fork. I've been creating some issues. Should I address them all in 1 PR or should I make separate PRs for each issue?

**Comment by rundef** [comment]
@NicholasBoll You can make just 1 PR, it's okay. Thanks ! :+1:

**Comment by NicholasBoll** [comment]
@rundef Thanks for taking a look!

**Pull request merged**